File: Orbit.Domain/DTOs/Pots/Responses/GetAllPotDataDto.cs
namespace Orbit.Domain.DTOs.Pots.Responses
{
    public class GetAllPotDataDto
    {
        public required SpendingPotData[] SpendingPots { get; set; }
        public required SavingsPotData[] SavingsPots { get; set; }
    }

    public class SpendingPotData
    {
        public required int PotId { get; set; }
        public required string PotName { get; set; }
        public required string AmountAllocated { get; set; }
        public required string AmountLeft { get; set; }
        public required string AmountSpent { get; set; }
    }

    public class SavingsPotData
    {
        public required int PotId { get; set; }
        public required string PotName { get; set; }
        public required string AmountSaved { get; set; }
        public required string AmountAddedThisMonth { get; set; }
    }
}

File: orbit.web/interfaces/api/pots/GetAllPotDataDto.ts
export interface GetAllPotDataDto {
  spendingPots: SpendingPotData[]
  savingsPots: SavingsPotData[]
}

export interface SpendingPotData {
  potId: number
  potName: string
  amountAllocated: string
  amountLeft: string
  amountSpent: string
}

export interface SavingsPotData {
  potId: number
  potName: string
  amountSaved: string
  amountAddedThisMonth: string
}
